crates/subspace-archiving/src/archiver.rs

```diff
@@ -647,7 +647,7 @@ impl<State: private::ArchiverState> Archiver<State> {
                 let witness = merkle_tree
                     .get_witness(position)
                     .expect("We use the same indexes as during Merkle tree creation; qed");
-                let mut piece: Piece = [0u8; PIECE_SIZE];
+                let mut piece = [0u8; PIECE_SIZE];
```

Comment:
With `AsMut` and `Default` impls, this could be simplified to:
```suggestion
                let mut piece = Piece::default();
```

without need to call `.into()` later here and in other places